crates/ruff_linter/src/rules/flake8_comprehensions/rules/unnecessary_comprehension_in_call.rs

```diff
@@ -29,22 +33,28 @@ use crate::rules::flake8_comprehensions::fixes;
 /// 212 ns ± 0.892 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)
 /// ```
 ///
-/// This performance difference is due to short-circuiting; if the entire iterable has to be
+/// This performance difference is due to short-circuiting. If the entire iterable has to be
 /// traversed, the comprehension version may even be a bit faster (list allocation overhead is not
-/// necessarily greater than generator overhead).
-///
-/// The generator version is more memory-efficient.
+/// necessarily greater than generator overhead). So applying this rule simplifies the code and
+/// will usually save memory, but in the absence of short-circuiting it may not improve (and may
+/// even slightly regress, though the difference will be small) performance.
```

Comment:
Quite a long sentence here; you could possibly split it into two:

```suggestion
/// necessarily greater than generator overhead). Applying this rule simplifies the code and
/// will usually save memory, therefore, but in the absence of short-circuiting it may
/// not improve performance. (In some situations, it *may* even slightly regress
/// performance, but the difference will usually be small.)
```